frontend/scripts/gameLogicInterface.js

```diff
@@ -285,7 +285,20 @@ export default function createGame(initialGameState = emptyGameState) {
 		 * Instantly drop the current tetromino as far as it goes and lock it in place
 		 */
 		instantDropTetromino: function() {
-
+			let this_row_valid = this.isStateValid(this.gameState.activeTetromino);
+
+			while (this_row_valid) {
+				this_row_valid = this.isStateValid(this.gameState.activeTetromino);
+				while (this_row_valid) {

```

Comment:
Why is there two while loops checking the same condition? Can you write this more nicely?